src/pages/home/report/ContextMenu/BaseReportActionContextMenu.tsx

```diff
@@ -0,0 +1,238 @@
+        contextAction.shouldShow(type, reportAction, isArchivedRoom, betas, anchor, isChronosReport, reportID, isPinnedChat, isUnreadChat, !!isOffline),
+    );
+
+    // Context menu actions that are not rendered as menu items are excluded from arrow navigation
+    const nonMenuItemActionIndexes = filteredContextMenuActions.map((contextAction, index) => (typeof contextAction.renderContent === 'function' ? index : undefined));
+    const disabledIndexes = nonMenuItemActionIndexes.filter((index): index is number => index !== undefined);
+
+    const [focusedIndex, setFocusedIndex] = useArrowKeyFocusManager({
+        initialFocusedIndex: -1,
+        disabledIndexes,
+        maxIndex: filteredContextMenuActions.length - 1,
+        isActive: shouldEnableArrowNavigation,
+    });
+
+    /**
+     * Checks if user is anonymous. If true and the action doesn't accept for anonymous user, hides the context menu and
+     * shows the sign in modal. Else, executes the callback.
+     */
+    const interceptAnonymousUser = (callback: () => void, isAnonymousAction = false) => {
+        if (Session.isAnonymousUser() && !isAnonymousAction) {
+            hideContextMenu(false);
+
+            InteractionManager.runAfterInteractions(() => {
+                Session.signOutAndRedirectToSignIn();
+            });
+        } else {
+            callback();
+        }
+    };
+
+    useKeyboardShortcut(
+        CONST.KEYBOARD_SHORTCUTS.ENTER,
+        (event) => {
+            if (!menuItemRefs.current[focusedIndex]) {
+                return;
+            }
+
+            // Ensures the event does not cause side-effects beyond the context menu, e.g. when an outside element is focused
+            if (event) {
+                event.stopPropagation();
+            }
+
+            menuItemRefs.current[focusedIndex].triggerPressAndUpdateSuccess();
+            setFocusedIndex(-1);
+        },
+        {isActive: shouldEnableArrowNavigation},
+    );
+
+    return (
+        (isVisible || shouldKeepOpen) && (
+            <View
+                ref={contentRef}
+                style={wrapperStyle}
+            >
+                {filteredContextMenuActions.map((contextAction, index) => {
+                    const closePopup = !isMini;
+                    const payload: ContextMenuActionPayload = {
+                        reportAction: reportAction as ReportAction,
+                        reportID,
+                        draftMessage,
+                        selection,
+                        close: () => setShouldKeepOpen(false),
+                        openContextMenu: () => setShouldKeepOpen(true),
+                        interceptAnonymousUser,
+                    };
+
+                    if (contextAction.renderContent) {
+                        // make sure that renderContent isn't mixed with unsupported props
+                        if (__DEV__ && contextAction.icon != null) {
```

Comment:
Why this part `contextAction.text != null `was removed?

Reply:
Because text doesn't exist on `ContextMenuAction`